### ISSUE TITLE
How to customize PHP DocBlocker ？

### ISSUE THREAD

**Issue by 377960738** [DEVSENSE/phptools-docs#157]
I want to customize information such as time and author, how to configure? Thanks.

**Comment by jakubmisek** [comment]
Hi, thanks for the feedback!

There are no configurable PHPDoc snippets yet, but we're planning to add them.

Are you referring to PHPDoc blocks above classes/interfaces, and at file beginnings?

**Comment by 377960738** [comment]
> Are you referring to PHPDoc blocks above classes/interfaces, and at file beginnings?

Yup. And the method.

**Comment by jakubmisek** [comment]
thanks! we'll add support for customizable PHPDoc snippets

**Comment by jakubmisek** [comment]
@377960738 there is a pre-release version `1.14.10470` which allows customizing the doc-comments snippets, i.e.:

```json
"php.docblock.classSnippet": {
  "@author": "John",
  "@copyright": "(c) $CURRENT_YEAR",
},
"php.docblock.functionSnippet": {
  "@throws": false,
}
```

Thank you for the feedback!

**Comment by tobsn** [comment]
@jakubmisek is $CURRENT_YEAR supposed to be replaced on auto complete? (because it isn't, v1.33.12973)

**Comment by jakubmisek** [comment]
@tobsn thank you for pointing this out.

It works when typing `/**`

![vsc-complete-phpdoc](https://user-images.githubusercontent.com/842150/232755552-2fb4bca4-ff36-4bae-867c-1117d80474a0.gif)

And it seems it's not completing the `$CURRENT_YEAR` when using a Code Action - will be fixed.

**Comment by tobsn** [comment]
@jakubmisek oh I see, yeah that is part of an sort of unrelated problem. when I type /** copilot tries to auto complete, which means I have to use the action on the sideline :D

**Comment by jakubmisek** [comment]
@tobsn good point :) we'll try to fix the code action .. in the future update, we'll try to handle `/**` differently so it would work with copilot hopefully...

**Comment by jakubmisek** [comment]
@tobsn I've implemented the phpdoc as a snippet for the code action as well. Thank you for reporting that.

The fix will be available in the next pre-release.
